meshkernel/utils.py

```diff
@@ -9,17 +15,20 @@ def plot_edges(node_x, node_y, edge_nodes, ax, *args, **kwargs):
         edge_nodes (ndarray, optional): A 1D integer array describing the nodes composing each mesh 2d edge.
         ax (matplotlib.axes.Axes): The axes where to plot the edges
     """
-    for edge_index in range(0, edge_nodes.size, 2):
-        first_edge_node_index = edge_nodes[edge_index]
-        second_edge_node_index = edge_nodes[edge_index + 1]
+    n_edge = int(edge_nodes.size / 2)
+    edge_coords = np.empty((n_edge, 2, 2), dtype=np.float64)
+    node_0 = edge_nodes[0::2]
+    node_1 = edge_nodes[1::2]
+    edge_coords[:, 0, 0] = node_x[node_0]
+    edge_coords[:, 0, 1] = node_y[node_0]
+    edge_coords[:, 1, 0] = node_x[node_1]
+    edge_coords[:, 1, 1] = node_y[node_1]
 
-        edge_x = [
-            node_x[first_edge_node_index],
-            node_x[second_edge_node_index],
+    if "colors" not in kwargs:
+        kwargs["colors"] = [
+            mcolors.to_rgba(c)
+            for c in plt.rcParams["axes.prop_cycle"].by_key()["color"]
```

Comment:
I personally don't really care about the old behavior, where it cycles through the colors per edge.
Unless you do, I think we can skip this block.

Reply:
Some of the node coordinates can have an invalid value (x= -999.0, y-999.0 ). We should prevent plotting these locations when using LineCollection.

Reply:
The original function does not have this behavior.

Taking a gander at the py_structures module, I don't take it currently supports invalid coordinates either. (https://github.com/Deltares/MeshKernelPy/blob/main/meshkernel/py_structures.py)

You'd have to store a sentinel value somewhere? The alternative in Python is using masked arrays, but I personally loathe them.

Reply:
Yeah, this is probably out of scope for this PR.
I will discuss with @lucacarniato where to handle this, because invalid values should indeed be handled at some point.